### PR TITLE
Find first error step based on "FinishAt" and "StartAt"

### PR DESCRIPTION
**Pull request by vincent-pli**
Fix issue: #2415

Introduce `StartAt` for the sorting when `FinishedAt` are exactly the same.
Since the goal is to find the first failed step, the StartAt and FinishedAt are most simple and directly solution.
Moreover, adopt a higher resolution format for `StartAt` to make it more accurately.

# Changes

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [ ] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/pipeline/blob/master/CONTRIBUTING.md) for more details._

Double check this list of stuff that's easy to miss:

- If you are adding [a new binary/image to the `cmd` dir](../cmd), please update
  [the release Task](../tekton/publish.yaml) to build and release this image.

## Reviewer Notes

If [API changes](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md) are included, [additive changes](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md#additive-changes) must be approved by at least two [OWNERS](https://github.com/tektoncd/pipeline/blob/master/OWNERS) and [backwards incompatible changes](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md#backwards-incompatible-changes) must be approved by [more than 50% of the OWNERS](https://github.com/tektoncd/pipeline/blob/master/OWNERS), and they must first be added [in a backwards compatible way](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md#backwards-compatible-changes-first).

### CODE REVIEW

File: pkg/entrypoint/entrypointer.go [hunk 1]
```diff
@@ -29,6 +29,11 @@ import (
 	"go.uber.org/zap"
 )
 
+//RFC3339 with millisecond
+const (
+	TimeFormat = "2006-01-02T15:04:05.000Z07:00"
```
Comment (sbwsg) [inline pkg/entrypoint/entrypointer.go:34]: no need to export this const i dont think?  `timeFormat`?

File: pkg/pod/status.go [hunk 2]
```diff
@@ -70,6 +70,9 @@ const (
 
 	// ReasonFailed indicates that the reason for the failure status is unknown or that one of the steps failed
 	ReasonFailed = "Failed"
+
+	//TimeFormat is RFC3339 with millisecond
+	TimeFormat = "2006-01-02T15:04:05.000Z07:00"
```
Comment (sbwsg) [inline pkg/pod/status.go:75]: same suggestion here, `timeFormat`

File: pkg/entrypoint/entrypointer.go (hunk 1 above)
Comment (vincent-pli) [reply pkg/entrypoint/entrypointer.go]: Done, thanks

File: pkg/pod/status.go (hunk 2 above)
Comment (vincent-pli) [reply pkg/pod/status.go]: Done, thanks

**Comment by sbwsg** [comment]
re-adding @vdemeester's lgtm

/lgtm

**Pull request merged**